File: src/main/java/com/example/tasksync/entities/Projeto.java
package com.example.tasksync.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;


@Entity
//Get and Setters
@Data
//Construtor sem argumentos
@NoArgsConstructor
//Construtor com argumentos
@AllArgsConstructor
public class Projeto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;

    private String descricao;

    private String status;

    private LocalDate dataInicio;

    private LocalDate dataPrevistaFim;

}

File: front/app/(sistema)/types/projeto.ts
export class Projeto {
  constructor(
    public id: number | null,
    public nome: string,
    public descricao: string,
    public dataInicio: string,
    public dataPrevistaFim: string,
    public status: string,
  ) {}
}

export interface ProjetoFormProps{
  projetoExistente?: Projeto
}







